db/migrate/20171006171936_add_data_download_version_per_context.rb

```diff
@@ -0,0 +1,5 @@
+class AddDataDownloadVersionPerContext < ActiveRecord::Migration[5.0]
+  def change
+    add_column :download_versions, :context_id, :integer
```

Comment:
in addition to adding the column we could already add rows for all contexts as part of migration and set the column as `NOT NULL`

Reply:
They requested that we also support contexts with no version, so we either make version optional at a database by having no row, or we need to allow null values. I believe no row is easier to maintain, otherwise it will require an explicit "empty" row, which is weird.

Reply:
I'd go with no row. Do we have the list of which versions / no versions to use and where?

Reply:
that's what we have. I also changed the logic that loads version to account for this.

current config here: https://basecamp.com/1756858/projects/12498794/todos/326870870